frame/bridge/relayer-game/src/lib.rs

```diff
@@ -187,66 +185,168 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		fn on_initialize(block_number: BlockNumber<T>) -> Weight {
-			if let Ok(weight) = Self::system_approve_pending_headers(block_number) {
-				weight
-			} else {
-				// TODO: handle error
+		fn offchain_worker(now: BlockNumber<T>) {
+			let game_ids = <ClosedRounds<T, I>>::take(now);
 
-				0
+			if !game_ids.is_empty() {
+				if let Err(e) = Self::update_games_at(game_ids, now) {
+					error!(target: "relayer-game", "{:?}", e);
+				}
 			}
+
+			// Return while no closed rounds found
 		}
 
-		// TODO: too many db operations and calc need to move to `offchain_worker`
-		// 	https://github.com/darwinia-network/darwinia-common/issues/254
-		// TODO: close the game that its id less than the best number
-		// 	https://github.com/darwinia-network/darwinia-common/issues/253
-		fn on_finalize(block_number: BlockNumber<T>) {
-			let closed_rounds = <ClosedRounds<T, I>>::take(block_number);
-
-			// `closed_rounds` MUST NOT be empty after this check; qed
-			if closed_rounds.len() != 0 {
-				// TODO: handle error
-				let _ = Self::settle(block_number, closed_rounds);
+		#[weight = 100_000_000]
+		pub fn update_games_unsigned(origin, game_ids: Vec<GameId<T, I>>) {
```

Comment:
```suggestion
		pub fn settle_games_unsigned(origin, game_ids: Vec<GameId<T, I>>) {
```

Reply:
Not just settle here

Would trigger a new round to update some games